### PR TITLE
modeltest-ng: adding new version, swapping maintainer

### PR DESCRIPTION
**Pull request by snehring**
Also allows for aarch64 compiles

### CODE REVIEW

**Review by bernhardkaindl — COMMENTED** [review]
Builds fine with these compilers on Rocky Linux 8.8 aarch64:
```py
rpm -qf /opt/rh/gcc-toolset-11/root/usr/bin/gcc /usr/bin/gcc
gcc-toolset-11-gcc-11.2.1-9.1.el8.aarch64
gcc-8.5.0-18.el8.aarch64
```
But the gcc-12.2.1 of gcc-toolset-12 of Rocky Linux 8.8 runs into this Internal Compiler Error (ICE):

Please add
```py
   # 40217: ICE by gcc-toolset-12-gcc of Rocky Linux 8.8:
   conflicts("%gcc@12.2.1", when="target=aarch64:")
```
unless you use a gcc-12.1.1 for aarch64 which does not have this ICE:
```py
     439    /tmp/bkaindl/spack-stage/spack-stage-modeltest-ng-20220721-4bl5be4kjdrkk2bdfjw52a4nrzwtaq34/spack-src/libs/pll-modules/libs/libpll/src/parsimony.c: In function 'pl
            l_parsimony_build':
  >> 440    /tmp/bkaindl/spack-stage/spack-stage-modeltest-ng-20220721-4bl5be4kjdrkk2bdfjw52a4nrzwtaq34/spack-src/libs/pll-modules/libs/libpll/src/parsimony.c:204:19: internal
             compiler error: in vect_transform_reduction, at tree-vect-loop.cc:7469
     441      204 | PLL_EXPORT double pll_parsimony_build(pll_parsimony_t * pars,
     442          |                   ^~~~~~~~~~~~~~~~~~~
     443    Please submit a full bug report, with preprocessed source.
     444    See <https://bugs.rockylinux.org/> for instructions.
```
```ml
rpm -qf /opt/rh/gcc-toolset-12/root/usr/bin/gcc
gcc-toolset-12-gcc-12.2.1-7.4.el8.aarch64
```

File: var/spack/repos/builtin/packages/modeltest-ng/package.py [hunk 1]
```diff
@@ -24,5 +25,9 @@ class ModeltestNg(CMakePackage):
     depends_on("flex", type="build")
     depends_on("openmpi", when="+mpi")
```
Comment (bernhardkaindl) [inline var/spack/repos/builtin/packages/modeltest-ng/package.py:27]: Please add a conflict for the ICE with gcc@12.2.1 on aarch64 (see my comment):
```suggestion
   # 40217: ICE by gcc-toolset-12-gcc-12.2.1-7.4.el8.aarch64 of Rocky Linux 8.8:
   conflicts("%gcc@12.2.1", when="target=aarch64:")
```

**Review by bernhardkaindl — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/modeltest-ng/package.py (hunk 1 above)
Comment (snehring) [reply var/spack/repos/builtin/packages/modeltest-ng/package.py:27]: Was able to reproduce this on alma. I don't readily have a real rhel box or any other distros installed on aarch64 to see if it's just the rebuilds or only rhel, but I'm fine with this.

**Comment by snehring** [comment]
oop, should have let my aarch64 with gcc@12.2.0 finish. That breaks too. Let me see if 12.3 fixes it.

**Review by bernhardkaindl — APPROVED** [review]

**Pull request merged**